File: backend/ebay.js
const eBayApi = require('ebay-api'); // Assume an eBay API SDK
const fs = require('fs');
const csvParser = require('csv-parser');
require('dotenv').config();

const devID = '811bfc1a-ccef-4627-82e3-ecc7bc652d71'

const eBay = new eBayApi({
    appId: process.env.CLIENT_ID,
    certId: process.env.CLIENT_SECRET,
    sandbox: false,
    siteId: eBayApi.SiteId.EBAY_US, // required for traditional APIs, see https://developer.ebay.com/DevZone/merchandising/docs/Concepts/SiteIDToGlobalID.html
    marketplaceId: eBayApi.MarketplaceId.EBAY_US, // default. required for RESTful APIs
    acceptLanguage: eBayApi.Locale.en_US, // default
    contentLanguage: eBayApi.Locale.en_US, // default.
});

eBay.OAuth2.setScope([
  'https://api.ebay.com/oauth/api_scope',
  'https://api.ebay.com/oauth/api_scope/sell.fulfillment.readonly',
  'https://api.ebay.com/oauth/api_scope/sell.fulfillment'
]);

// 2. Generate and open Url and Grant Access
const url = eBay.OAuth2.generateAuthUrl();
console.log('Open URL', url);

//Your eBay API credentials


// const createListing = (item) => {
//     eBayApi.addFixedPriceItem({
//         title: item.Title,
//         description: item.Description,
//         price: item.Price,
//         category_id: item.CategoryID,
//         condition_id: item.ConditionID,
//         quantity: item.Quantity,
//         location: {
//             country: item.Country,
//             postal_code: item.PostalCode,
//             location: item.Location
//         },
//         shipping_details: {
//             shipping_service: item.ShippingService,
//             shipping_cost: item.ShippingCost
//         }
//     }).then((response) => {
//         console.log('Listing created:', response);
//     }).catch((error) => {
//         console.error('Error creating listing:', error);
//     });
// };

// const listItemsFromCSV = (csvFilePath) => {
//     const items = [];

//     fs.createReadStream(csvFilePath)
//         .pipe(csvParser())
//         .on('data', (row) => {
//             items.push(row);
//         })
//         .on('end', () => {
//             console.log('CSV file successfully processed');
//             items.forEach(createListing);
//         });
// };

//listItemsFromCSV('./items.csv');

File: backend/ebay.py
import requests
import json

# eBay endpoint for creating a listing
url = "https://api.ebay.com/sell/inventory/v1/inventory_item"

# ADD YOUR TOKENS AND AUTH
headers = {
    "Content-Type": "application/json",
    "Authorization": "Bearer",
}

item_data = {
    "sku": "sku-123",
    "offer": {
        "categoryId": "12345",
        "format": "FIXED_PRICE", 
        "listingDescription": "test",
        "marketplaceId": "EBAY_US", 
        "price": {
            "currency": "USD",
            "value": "25000.00"
        },
        "quantityLimitPerBuyer": 1
    },
    "product": {
        "title": "BLAH BLAH Item",
        "description": "VERY BLAH BLAH",
        "imageUrls": ["http://example.com/image1.jpg", "http://example.com/image2.jpg"],
    }
}

# POST IT!!!!!!!!!!!!
response = requests.post(url, headers=headers, data=json.dumps(item_data))

if response.ok:
    print("Listing created successfully.")
else:
    print(f"Failed to create listing. Response code: {response.status_code}, Detail: {response.text}")
